onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -609,6 +609,23 @@ TEST(GraphTransformationTests, Gemm_Relu_three_input) {
   std::map<std::string, int> op_to_count = CountOpsInGraph(graph);
   ASSERT_TRUE(op_to_count["Relu"] == 0);
 }
+
+TEST(GraphTransformationTests, Gemm_LeakyRelu_Fusion) {
+  auto model_uri = MODEL_FOLDER "gemm_activation_fusion/gemm_activation_fusion.onnx";
+
+  std::shared_ptr<Model> p_model;
+  ASSERT_TRUE(Model::Load(model_uri, p_model, nullptr, DefaultLoggingManager().DefaultLogger()).IsOK());
+  Graph& graph = p_model->MainGraph();
+  std::map<std::string, int> op_to_count1 = CountOpsInGraph(graph);
+  onnxruntime::GraphTransformerManager graph_transformation_mgr{5};
+  graph_transformation_mgr.Register(onnxruntime::make_unique<GemmActivationFusion>(), TransformerLevel::Level2);
+  ASSERT_TRUE(graph_transformation_mgr.ApplyTransformers(graph, TransformerLevel::Level2, DefaultLoggingManager().DefaultLogger()).IsOK());
+
+  std::map<std::string, int> op_to_count = CountOpsInGraph(graph);
+  ASSERT_TRUE(op_to_count["LeakyRelu"] == 0);
+  ASSERT_TRUE(op_to_count["Gemm"] == 0);
+  ASSERT_TRUE(op_to_count["FusedGemm"] == 1);
+}
```

Comment:
This test would have passed without your bug fix though wouldn't it? Should the test also check the attributes of the FusedGemm node are correct to prevent any regressions?

Reply:
Nope this test does not pass without my bug fix... 
After applying the transformer since modified is set to true graph.Resolve() is called from GraphTransformer::Apply mtd and it returns an error "This is an invalid model. Error in Node:fused _0 : Attribute 'alpha' appeared multiple times."